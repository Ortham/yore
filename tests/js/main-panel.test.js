import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../src/gui/js/map-area', () => 'MapArea');

import MainPanel from '../../src/gui/js/main-panel';

function apply() {
    return 'apply';
}

function discard() {
    return 'discard';
}

describe('MainPanel', () => {
    test('renders an image, map, location description and disabled buttons if photo location is not suggested', () => {
        let photo = {
            src: 'path',
            location: {
                Existing: {
                    latitude: 52.0,
                    longitude: 36.2
                }
            }
        };
        const mainPanel = renderer.create(
            <MainPanel photo={photo}
                handleSuggestionApply={apply}
                handleSuggestionDiscard={discard}
            />
        ).toJSON();
        expect(mainPanel).toMatchSnapshot();
    });

    test('renders an image, map, text and disabled buttons if photo has no location', () => {
        let photo = {
            src: 'path'
        };
        const mainPanel = renderer.create(
            <MainPanel photo={photo}
                handleSuggestionApply={apply}
                handleSuggestionDiscard={discard}
            />
        ).toJSON();
        expect(mainPanel).toMatchSnapshot();
    });

    test('renders an image, map, location description and enabled buttons if photo location is suggested', () => {
        let photo = {
            src: 'path',
            location: {
                Suggested: [{
                    latitude: 52.0,
                    longitude: 36.2
                },{
                    meters: 5,
                    seconds: 10
                }]
            }
        };
        const mainPanel = renderer.create(
            <MainPanel photo={photo}
                handleSuggestionApply={apply}
                handleSuggestionDiscard={discard}
            />
        ).toJSON();
        expect(mainPanel).toMatchSnapshot();
    });

    test('renders text and disabled buttons if photo is undefined', () => {
        const mainPanel = renderer.create(<MainPanel />).toJSON();
        expect(mainPanel).toMatchSnapshot();
    });
});
