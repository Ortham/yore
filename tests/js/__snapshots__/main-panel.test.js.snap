// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPanel renders an image, map, location description and disabled buttons if photo location is not suggested 1`] = `
<main>
  <section>
    <img
      src="path"
    />
    <MapArea
      photo={
        Object {
          "location": Object {
            "Existing": Object {
              "latitude": 52,
              "longitude": 36.2,
            },
          },
          "src": "path",
        }
      }
    />
  </section>
  <footer>
    Existing location
    <br />
    <button
      disabled={true}
      onClick={[Function]}
    >
      Apply
    </button>
    <button
      disabled={true}
      onClick={[Function]}
    >
      Discard
    </button>
  </footer>
</main>
`;

exports[`MainPanel renders an image, map, location description and enabled buttons if photo location is suggested 1`] = `
<main>
  <section>
    <img
      src="path"
    />
    <MapArea
      photo={
        Object {
          "location": Object {
            "Suggested": Array [
              Object {
                "latitude": 52,
                "longitude": 36.2,
              },
              Object {
                "meters": 5,
                "seconds": 10,
              },
            ],
          },
          "src": "path",
        }
      }
    />
  </section>
  <footer>
    Suggested location: accuracy is 5 meters and 10 seconds
    <br />
    <button
      disabled={false}
      onClick={[Function]}
    >
      Apply
    </button>
    <button
      disabled={false}
      onClick={[Function]}
    >
      Discard
    </button>
  </footer>
</main>
`;

exports[`MainPanel renders an image, map, text and disabled buttons if photo has no location 1`] = `
<main>
  <section>
    <img
      src="path"
    />
    <MapArea
      photo={
        Object {
          "src": "path",
        }
      }
    />
  </section>
  <footer>
    No location
    <br />
    <button
      disabled={true}
      onClick={[Function]}
    >
      Apply
    </button>
    <button
      disabled={true}
      onClick={[Function]}
    >
      Discard
    </button>
  </footer>
</main>
`;

exports[`MainPanel renders text and disabled buttons if photo is undefined 1`] = `
<main>
  <section>
    No photo selected.
  </section>
  <footer>
    <br />
    <button
      disabled={true}
    >
      Apply
    </button>
    <button
      disabled={true}
    >
      Discard
    </button>
  </footer>
</main>
`;
